packages/botonic-nlp/src/tasks/ner/process/processor.ts

```diff
@@ -0,0 +1,97 @@
+import { tensor, Tensor2D } from '@tensorflow/tfjs-node'
+
+import { Sample } from '../../../dataset/types'
+import { Codifier } from '../../../preprocess/codifier'
+import { PADDING_TOKEN, UNKNOWN_TOKEN } from '../../../preprocess/constants'
+import { Preprocessor } from '../../../preprocess/preprocessor'
+import { NEUTRAL_ENTITY } from './constants'
+import { InputData, OutputData } from './types'
+
+export class Processor {
+  constructor(
+    public preprocessor: Preprocessor,
+    readonly tokenCodifier: Codifier,
+    readonly entityCodifier: Codifier
+  ) {}
+
+  // Processes samples and generates the Input and Output data.
+  process(samples: Sample[]): { x: InputData; y: OutputData } {
+    const processedSamples = samples.map(s => this.processSample(s))
+    return {
+      x: tensor(processedSamples.map(s => s.x)),
+      y: tensor(processedSamples.map(s => s.y)),
+    }
+  }
+
+  // Generates the Input data and the unmasked sequence.
+  generateInput(text: string): { sequence: string[]; input: InputData } {
+    const sequence = this.processText(text)
+    const input = tensor([this.processTokens(sequence)]) as Tensor2D
+    return { sequence, input }
+  }
+
+  private processSample(sample: Sample): { x: number[]; y: number[][] } {
+    let tokens: string[] = []
```

Comment:
this function is very long. Can it be split?

Reply:
I've been thinking about it but I don't find a way to correctly split it. I keep it like this for the moment.